src/mmrelay/matrix_utils.py

```
@@ -3889,26 +3892,46 @@ async def on_decryption_failure(room: MatrixRoom, event: MegolmEvent) -> None:
+                f"Requested keys for failed decryption of event {event.event_id} "
+                f"(attempt {attempt + 1}/{E2EE_KEY_REQUEST_MAX_RETRIES})"
             )
-            return
-        request = event.as_key_request(matrix_client.user_id, matrix_client.device_id)
-        await matrix_client.to_device(request)
-        logger.info(f"Requested keys for failed decryption of event {event.event_id}")
-    except NIO_COMM_EXCEPTIONS:
-        logger.exception(f"Failed to request keys for event {event.event_id}")
+            # Success - wait once for key to arrive via federation
+            await asyncio.sleep(E2EE_KEY_REQUEST_BASE_DELAY)
+            return  # Exit after successful request
```

Ruff (0.15.5)

[warning] Consider moving this statement to an `else` block

(TRY300)